src/typeDefs/types.ts

```diff
@@ -294,6 +295,15 @@ export const types = gql`
     aggregate: AggregatePost!
   }
 
+  type Tag {
+    _id: ID!
+    title: String!
+    organization: Organization!
+    parentTag: Tag
+    childTags: [Tag]
```

Comment:
I am providing support for these as the front-end might need to display the hierarchy of tags. So in a certain view, they might want to query for the parent tag of the current tag, or the child ones. 
Providing support for the same is a good pattern in my opinion, as we are not storing any additional data for the same. They would be implemented using field-resolvers, thus would only run when specially queried for.

Reply:
Hmm, unless you're building some kind of sophisticated UI design I don't see where this'll be used. The tags for an organization will just be fetched as a list, then the UI will be rendered according to some function logic which will render each tag successively according to parentTagId and the UI behaviour will also depend on parentId(for example adding padding to the left of tags, for each successive child tag).

I personally think it'll be never used, and it's a very niche feature, but if you want to provide this functionality then sure.

Reply:
Nevermind, you're correct. If the tags are to be lazily loaded on the client side, this will be a required. But this feature can only truly be used when there's also a query field for a tag or a node interface query field.

I just thought maybe returning all tags in organization's tags field is a bad idea. We should only return the tags that have `parentId === null` for that organization. Child tags should always be lazily loaded. What do you think?

Reply:
Okay got your concerns!